### ISSUE TITLE
There is something error in the code!

### ISSUE THREAD

**Issue by GuideWsp** [saeed-anwar/DRLN#5]
Don't any one find the error in the code?
https://github.com/saeed-anwar/DRLN/blob/master/TestCode/code/model/drln.py#L144
        b4 = self.b4(a1)
        c4 = torch.cat([o3, b4], dim=1)
        o4 = self.c4(c4)
 
        b5 = self.b5(a1)                                # is it should be bt = self.b5(04)
        c5 = torch.cat([c4, b5], dim=1)        # is it should be c5 = torch.cat([o4, c5], dim=1)
        o5 = self.c5(c5)

        b6 = self.b6(o5)
        c6 = torch.cat([c5, b6], dim=1)
        o6 = self.c6(c6)
        a2 = o6 + a1

Hoping for the author's relpy.

**Comment by wch1996** [comment]
Not only the code have some errors,but also this paper have some problems.For example,this paper proposed laplace attention machanism,it is fake,just a simple SE

**Comment by GuideWsp** [comment]
Yes, I have tried it,the laplace attention can be converted to SE.

**Comment by waleedrazakhan92** [comment]
Did anyone successfully run the code? I'm unable to run even the inference. No idea where to expect the sharpened images. I'm using torch 1.1.0 and torchvision 0.4.0. The code doesn't run with other combinations of torch and torchvision so i'm stuck with these. There's a problem, something to do with it being 'not divisible by zero' with self.loader something. Please guide @GuideWsp @wch1996 @saeed-anwar

**Comment by saeed-anwar** [comment]
> Don't any one find the error in the code?
> https://github.com/saeed-anwar/DRLN/blob/master/TestCode/code/model/drln.py#L144
> b4 = self.b4(a1)
> c4 = torch.cat([o3, b4], dim=1)
> o4 = self.c4(c4)
> 
> ```
>     b5 = self.b5(a1)                                # is it should be bt = self.b5(04)
>     c5 = torch.cat([c4, b5], dim=1)       
>     o5 = self.c5(c5)
> 
>     b6 = self.b6(o5)
>     c6 = torch.cat([c5, b6], dim=1)
>     o6 = self.c6(c6)
>     a2 = o6 + a1
> ```
> 
> Hoping for the author's relpy.

Hi, 
Sorry for the delayed response. Unfortunately, I don't get much time to respond to the queries or question.
We are aware of the connection skip here. However, it is not going to effect the performance, the block which is skipped by the connection is not contributing anything in the model. If that block is included in the model it will further increase the performance. Hope this helps. 

Hopefully, in the future if the time permits we will retrain the model and put it out there including the skipped block.

**Comment by saeed-anwar** [comment]
> Not only the code have some errors,but also this paper have some problems.For example,this paper proposed laplace attention machanism,it is fake,just a simple SE

Hi,

Laplacian attention is the idea employing the SE block. Please read the paper properly. SE can be replaced by any attentional block. We have provided comparisons in the supplementary materials between vanilla SE block and Laplacian attention using SE block.

Hope this helps.

**Comment by saeed-anwar** [comment]
> Did anyone successfully run the code? I'm unable to run even the inference. No idea where to expect the sharpened images. I'm using torch 1.1.0 and torchvision 0.4.0. The code doesn't run with other combinations of torch and torchvision so i'm stuck with these. There's a problem, something to do with it being 'not divisible by zero' with self.loader something. Please guide @GuideWsp @wch1996 @saeed-anwar

Hi,

Sorry to hear that.  The SR images will be in the directory you specify in the command line. Search for the save directory in the folder you are running the code.

**Comment by Akhtar303** [comment]
> > Did anyone successfully run the code? I'm unable to run even the inference. No idea where to expect the sharpened images. I'm using torch 1.1.0 and torchvision 0.4.0. The code doesn't run with other combinations of torch and torchvision so i'm stuck with these. There's a problem, something to do with it being 'not divisible by zero' with self.loader something. Please guide @GuideWsp @wch1996 @saeed-anwar
> 
> Hi,
> 
> Sorry to hear that. The SR images will be in the directory you specify in the command line. Search for the save directory in the folder you are running the code.

@saeed-anwar @waleedrazakhan92 
I am facing same issue.
I have run successfully on my local machine(ubuntu 18.04) but doesn't run my server machine(ubuntu 20.0)  with torch 1.1.0 and torchvision 0.3.0 with cuda 9.0 and cudnn 7.1.2 same version on both ubuntu machine .
Error "'not divisible by zero" issue with Dataloader .
@waleedrazakhan92 have you resolve this isuue.
Thanks

**Comment by Akhtar303** [comment]
> Did anyone successfully run the code? I'm unable to run even the inference. No idea where to expect the sharpened images. I'm using torch 1.1.0 and torchvision 0.4.0. The code doesn't run with other combinations of torch and torchvision so i'm stuck with these. There's a problem, something to do with it being 'not divisible by zero' with self.loader something. Please guide @GuideWsp @wch1996 @saeed-anwar

@waleedrazakhan92 
Solution :
pip install scipy==1.1.0
the main issue is that the scipy.misc is removed after scipy 1.2.0 and that's why the images are not being read and it was returning "not divisible by zero"

**Comment by dynamiterdx** [comment]
Also, _DataLoaderIter has been removed from PyTorch 1.3 onwards.
